nscplugin/src/main/scala/scala/scalanative/nscplugin/NirGenExpr.scala

```diff
@@ -1504,10 +1520,11 @@ trait NirGenExpr { self: NirGenPhase =>
       val ty       = genType(st)
       val args     = genSimpleArgs(argsp)
       var res: Val = Val.Zero(ty)
+      val argspVec = argsp.toVector
```

Comment:
Consider `zip`ing over `argsp` in addition to the `zipWithIndex` so that you don't have to create this copy of `argsp`.